File: gommo-frontend/src/modules/person/collaborators/people/lib/collaborator.mapper.ts
import type {Collaborator, CollaboratorCreateDto} from "@/modules/person/collaborators/people/dto/collaborator.dto";
import {digitsOnly} from "@/shared/lib/input/digits";

export function collaboratorFormToPayload(dto: CollaboratorCreateDto): CollaboratorCreateDto {
    return {
        ...dto,
        cpf: digitsOnly(dto.cpf),
        rg: dto.rg ? digitsOnly(dto.rg).slice(0, 7) : dto.rg,
        rgIssuer: dto.rgIssuer?.trim() || undefined,
        rgStateCode: dto.rgStateCode?.trim().toUpperCase() || undefined,
    };
}

export function collaboratorToFormDto(collaborator: Collaborator): CollaboratorCreateDto {
    return {
        fullName: collaborator.fullName,
        socialName: collaborator.socialName,
        cpf: digitsOnly(collaborator.cpf),
        rg: collaborator.rg ? digitsOnly(collaborator.rg).slice(0, 7) : collaborator.rg,
        rgIssuer: collaborator.rgIssuer ?? "",
        rgStateCode: collaborator.rgStateCode ?? "",
        birthDate: collaborator.birthDate?.slice(0, 10) ?? "",
        gender: collaborator.gender,
        maritalStatus: collaborator.maritalStatus,
        motherName: collaborator.motherName,
        fatherName: collaborator.fatherName,
        nationality: collaborator.nationality,
        pisPasep: collaborator.pisPasep,
    };
}

export const emptyCollaboratorForm = (): CollaboratorCreateDto => ({
    fullName: "",
    cpf: "",
    birthDate: "",
});

File: gommo-backend/src/main/java/br/com/gommo/modules/person/collaborators/people/mapper/CollaboratorMapper.java
package br.com.gommo.modules.person.collaborators.people.mapper;

import br.com.gommo.modules.person.collaborators.people.dto.CollaboratorRequestDto;
import br.com.gommo.modules.person.collaborators.people.dto.CollaboratorResponseDto;
import br.com.gommo.modules.person.collaborators.people.entity.Collaborator;
import org.springframework.stereotype.Component;

@Component
public class CollaboratorMapper {

    public Collaborator toEntity(CollaboratorRequestDto dto) {
        return Collaborator.builder()
                .fullName(dto.getFullName())
                .socialName(dto.getSocialName())
                .cpf(dto.getCpf())
                .rg(dto.getRg())
                .rgIssuer(dto.getRgIssuer())
                .rgStateCode(dto.getRgStateCode())
                .birthDate(dto.getBirthDate())
                .gender(dto.getGender())
                .maritalStatus(dto.getMaritalStatus())
                .motherName(dto.getMotherName())
                .fatherName(dto.getFatherName())
                .nationality(dto.getNationality() != null ? dto.getNationality() : "Brasileiro")
                .pisPasep(dto.getPisPasep())
                .photoObjectId(dto.getPhotoObjectId())
                .build();
    }

    public void updateEntity(Collaborator entity, CollaboratorRequestDto dto) {
        entity.setFullName(dto.getFullName());
        entity.setSocialName(dto.getSocialName());
        entity.setCpf(dto.getCpf());
        entity.setRg(dto.getRg());
        entity.setRgIssuer(dto.getRgIssuer());
        entity.setRgStateCode(dto.getRgStateCode());
        entity.setBirthDate(dto.getBirthDate());
        entity.setGender(dto.getGender());
        entity.setMaritalStatus(dto.getMaritalStatus());
        entity.setMotherName(dto.getMotherName());
        entity.setFatherName(dto.getFatherName());
        if (dto.getNationality() != null) {
            entity.setNationality(dto.getNationality());
        }
        entity.setPisPasep(dto.getPisPasep());
        entity.setPhotoObjectId(dto.getPhotoObjectId());
    }

    public CollaboratorResponseDto toResponse(Collaborator entity) {
        return toResponse(entity, null, null);
    }

    public CollaboratorResponseDto toResponse(Collaborator entity, String email, String phone) {
        return CollaboratorResponseDto.builder()
                .id(entity.getId())
                .code(entity.getCode())
                .status(entity.getStatus())
                .fullName(entity.getFullName())
                .socialName(entity.getSocialName())
                .cpf(entity.getCpf())
                .rg(entity.getRg())
                .rgIssuer(entity.getRgIssuer())
                .rgStateCode(entity.getRgStateCode())
                .birthDate(entity.getBirthDate())
                .gender(entity.getGender())
                .maritalStatus(entity.getMaritalStatus())
                .motherName(entity.getMotherName())
                .fatherName(entity.getFatherName())
                .nationality(entity.getNationality())
                .pisPasep(entity.getPisPasep())
                .email(email)
                .phone(phone)
                .photoObjectId(entity.getPhotoObjectId())
                .createdAt(entity.getCreatedAt())
                .updatedAt(entity.getUpdatedAt())
                .build();
    }
}
